### PR TITLE
Enhance command line processing

### PR DESCRIPTION
**Pull request by joejulian**
Just some simple command line option parsing changes to support long options and add some help text.

Also a minor change to allow a manually configured MAPPINGS environment variable to be used to manually add additional `-v source:dest` docker switches.

### CODE REVIEW

File: hack/dockerdev [hunk 1]
```diff
@@ -23,11 +36,18 @@ EOF
...
 fi
 
-while getopts c:i:r: opt; do
-    case $opt in
-        c) CONFIGFILE="${OPTARG}";;
-        i) IMAGE="${OPTARG}";;
-        r) MYHOME="${OPTARG}";;
+# Parse command line options
+GETOPT=$(getopt -q -o c:fhi:m: --long config:,fresh,help,image:,home: -n $0 -- "$@")
+eval set -- "$GETOPT"
```
Comment (mikeln) [inline hack/dockerdev]: fresh?   not mentions in the usage or the parsing.

File: hack/dockerdev [hunk 2]
```diff
@@ -48,7 +68,7 @@ if ! [[ -f ${CONFIGFILE} ]]; then
 fi
 
 CONFIGFILE=$(realpath ${CONFIGFILE})
-MAPPINGS="-v ${CONFIGFILE}:${CONFIGFILE}"
+MAPPINGS="$MAPPINGS -v ${CONFIGFILE}:${CONFIGFILE}"
```
Comment (mikeln) [inline hack/dockerdev]: so you are allowing MAPPINGS brought in from the environment?

Comment (joejulian) [inline hack/dockerdev]: Yes, it came up a couple weeks ago when someone wanted to map a particular file through to their dev environment. It can be done just by adding dummy entries to the config file, but this seemed less hacky.

File: hack/dockerdev (hunk 1 above)
Comment (joejulian) [inline hack/dockerdev]: That's what I get for doing three changes at once, then realizing I should have done them as three separate PRs. :grin:
